src/test/libcephfs/monaddrsfromcct.cc

```diff
@@ -0,0 +1,103 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2020 Red Hat
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "gtest/gtest.h"
+#include "include/cephfs/libcephfs.h"
+#include "common/async/context_pool.h"
+#include "common/ceph_context.h"
+#include "mon/MonClient.h"
+#include <errno.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <dirent.h>
+#ifdef __linux__
+#include <sys/xattr.h>
+#endif
+
+// Helper to remove/unset all possible mon information from ConfigProxy
+void helper_clear_cct_of_mon_config(CephContext *cct) {
+  auto& conf = cct->_conf;
+  if (const auto monmap = conf.get_val<std::string>("monmap");
+    !monmap.empty()) {
+    ASSERT_EQ(0, conf.set_val("monmap", "", nullptr));
+  }
+  if (const auto mon_host = conf.get_val<std::string>("mon_host");
+      !mon_host.empty()) {
+    ASSERT_EQ(0, conf.set_val("mon_host", "", nullptr));
+  }
+  if (const auto mon_dns_name = conf.get_val<std::string>("mon_dns_srv_name");
+      !mon_dns_name.empty()) {
+    ASSERT_EQ(0, conf.set_val("mon_dns_srv_name", "", nullptr));
+  }
+}
+
+TEST(LibCephFS, MonaddrsFromCct) {
+  struct ceph_mount_info *ca, *cb;
+  CephContext *cct;
+  ceph::async::io_context_pool icp;
+
+  // Test mount failure when there is no known mon config source
```

Comment:
is it possible to break this into smaller tests?

Reply:
Done